.github/workflows/build.yml

```diff
@@ -9,31 +9,78 @@ on:
     types:
       - opened
       - synchronize
+
 concurrency:
   group: build-${{ github.ref }}
   cancel-in-progress: true
 
 jobs:
-  haskell:
+  build:
+    strategy:
+      matrix:
+        runner:
+          - macos-latest
+          - ubuntu-latest
+    runs-on: ${{ matrix.runner }}
+    env:
+      LANG: en_US.UTF-8
+      LC_ALL: en_US.UTF-8
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Install Nix
+        uses: cachix/install-nix-action@v31
+      - name: Setup Cachix
+        uses: cachix/cachix-action@v16
+        with:
+          name: devenv
+      - name: Install devenv
+        run: nix profile install nixpkgs#devenv 
+      - name: Setup cache for `~/.stack`
+        uses: actions/cache@v4
+        with:
+          path: ~/.stack
+          key: ${{ runner.os }}-stack-global-${{ hashFiles('stack.yaml') }}-${{ hashFiles('hapistrano.cabal') }}
+          restore-keys: |
+            ${{ runner.os }}-stack-global-${{ hashFiles('stack.yaml') }}-
+            ${{ runner.os }}-stack-global-
+      - name: Install dependencies
+        run: devenv shell -- stack build --only-dependencies --test
```

Comment:
Do we need `--test` for dependencies?

Reply:
@DavidMazarro Yes! We need it to build the test dependencies as well.